### PR TITLE
Migrate `clinical-data` endpoint to clean architecture with ClickHouse support

### LINKED ISSUE #11384: Clickhouse-ify clinical-data endpoint
https://genie.cbioportal.org/api/clinical-data/fetch?clinicalDataType=PATIENT

### PR DESCRIPTION
**Pull request by fuzhaoyuan**
Fix #11384

# Migrate `clinical-data` to Clean Architecture with ClickHouse Support

## Overview
This PR refactors clinical data endpoints from legacy architecture to clean architecture with ClickHouse support, as part of the broader backend migration to ClickHouse and clean architecture patterns.

### Domain Layer - Record-Based Models
- **ClinicalData Record**: Immutable record with multiple constructor overloads for different projection levels (ID, SUMMARY, DETAILED)
- **ClinicalAttribute Record**: Nested domain object for clinical attribute metadata

### Use Case Layer 
- **GetClinicalDataUseCase**: Main business logic with projection-based routing
- **GetClinicalDataMetaUseCase**: Dedicated metadata/count operations

### Repository Layer - Projection-Based Queries
New repository methods supporting different data retrieval levels:
- `getClinicalDataId()` - Basic identifiers only
- `getClinicalDataSummary()` - Identifiers + attribute values  
- `getClinicalDataDetailed()` - Complete data with clinical attribute metadata
- `getClinicalDataMeta()` - Count operations only

### MyBatis Layer - Constructor Mapping
- Replaced `<result>` + `<association>` patterns with `<constructor>` mapping for Records
- Separate ResultMaps and queries for each projection level
- Fixed package name issue: `clinical_data` → `clinical_attributes`

### REST Layer
- **ColumnarStoreClinicalDataController**: New `@Profile("clickhouse")` controller supporting multiple projection types
- **DTO Layer**: Response DTOs with MapStruct transformation including computed fields

### MapStruct Integration
- **ClinicalDataMapper**: Main mapper with computed `uniqueKey` fields
- **ClinicalAttributeMapper**: Nested object transformation  
- Automatic nested mapping via `uses = ClinicalAttributeMapper.class`

### Core New Architecture
- `domain/clinical_data/ClinicalData.java`
- `domain/clinical_attributes/ClinicalAttribute.java` 
- `domain/clinical_data/repository/ClinicalDataRepository.java`
- `domain/clinical_data/usecase/GetClinicalDataUseCase.java`
- `domain/clinical_data/usecase/GetClinicalDataMetaUseCase.java`
- `application/rest/vcolumnstore/ColumnarStoreClinicalDataController.java`
- `application/rest/response/ClinicalDataDTO.java`
- `application/rest/response/ClinicalAttributeDTO.java`
- `application/rest/mapper/ClinicalDataMapper.java`
- `application/rest/mapper/ClinicalAttributeMapper.java`
- MyBatis XML mapper (complete ResultMap and query restructuring)

### Compatibility Updates
- `domain/clinical_data/usecase/ClinicalDataUseCases.java`
- `domain/clinical_data/usecase/GetPatientClinicalDataUseCase.java`
- `domain/clinical_data/usecase/GetSampleClinicalDataUseCase.java` 
- `domain/clinical_data/usecase/GetClinicalDataForXyPlotUseCase.java`
- `domain/clinical_data/ClinicalDataType.java`
- `application/rest/vcolumnstore/ColumnarStoreStudyViewController.java`

## Testing
Added comprehensive integration tests in `ClickhouseClinicalDataMapperTest`:
- **testDataCountConsistency()**: Verifies data retrieval counts match metadata counts across different filters
- **testEmptyAndNullConditions()**: Tests edge cases with non-existent IDs and attributes
- **testProjectionTypesConsistency()**: Validates that different projection levels return consistent record counts while providing appropriate field sets

## Migration Notes
- New endpoints are isolated under `@Profile("clickhouse")`
- Legacy endpoints remain unchanged
- No breaking changes - new API returns consistent data format with existing endpoints
- Part of broader ClickHouse and clean architecture migration

### CODE REVIEW

File: src/main/java/org/cbioportal/legacy/service/util/ClinicalDataUtil.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package org.cbioportal.legacy.service.util;
+
+import java.util.Collections;
+import java.util.List;
+import org.cbioportal.legacy.model.ClinicalData;
+
+public class ClinicalDataUtil {
+  private ClinicalDataUtil() {}
+
+  public static ClinicalData convertToLegacyClinicalData(
```
Comment (alisman) [inline src/main/java/org/cbioportal/legacy/service/util/ClinicalDataUtil.java:18]: @fuzhaoyuan could we have a comment here explaining why we need to do this conversion?

File: src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnarStoreStudyViewController.java [hunk 2]
```diff
@@ -301,9 +302,10 @@ public ResponseEntity<DensityPlotData> fetchClinicalDataDensityPlot(
             .yAxisLogScale(yAxisLogScale)
             .build();
 
-    List<ClinicalData> combinedClinicalDataList =
-        studyViewService.getClinicalDataForXyPlot(
-            studyViewFilter, List.of(xAxisAttributeId, yAxisAttributeId), false);
+    List<org.cbioportal.legacy.model.ClinicalData> combinedClinicalDataList =
```
Comment (haynescd) [inline src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnarStoreStudyViewController.java:305]:
```suggestion
    List<ClinicalData> combinedClinicalDataList =
```

File: src/main/java/org/cbioportal/legacy/service/util/ClinicalDataUtil.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package org.cbioportal.legacy.service.util;
+
+import java.util.Collections;
+import java.util.List;
+import org.cbioportal.legacy.model.ClinicalData;
+
+public class ClinicalDataUtil {
```
Comment (haynescd) [inline src/main/java/org/cbioportal/legacy/service/util/ClinicalDataUtil.java]:
```suggestion
public abstract class ClinicalDataUtil {
```

Prevents users from instantiating the class

File: src/main/java/org/cbioportal/legacy/service/util/ClinicalDataUtil.java (hunk 1 above)
Comment (fuzhaoyuan) [reply src/main/java/org/cbioportal/legacy/service/util/ClinicalDataUtil.java:18]: Yes I'll add a comment

File: src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnarStoreStudyViewController.java (hunk 2 above)
Comment (fuzhaoyuan) [reply src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnarStoreStudyViewController.java:305]: This is because these old service methods are still using old models. To replace all usage with new models is too much of a overhaul so I made some temporary conversion util methods for them.

File: src/main/java/org/cbioportal/legacy/service/util/ClinicalDataUtil.java (hunk 3 above)
Comment (fuzhaoyuan) [reply src/main/java/org/cbioportal/legacy/service/util/ClinicalDataUtil.java]: Updated. Thanks!
